### PR TITLE
Monsterhunter rework

### PR DESCRIPTION
**Pull request by SmoSmoSmoSmok**
TL:DR monster hunter KILLS monsters he hunts mosnters and then kills themm then he killed them and they are died because they got killed by the monsnster killer he is a hunter and hunted them

<details>
<summary>Lore</summary>
<br>
The monster hunters are a private military company, travelling from planet to planet fighting wars for the side that pays higher. Over the years, continuous wars have taken its toll on the company. With a fraction of their original budget and most of their warriors dead the organization has become a shell of its former self. Their desperation to restore their former glory drove them to seek out new unorthodox strategies of warfare. THUS, their research team was able to discover a distant moon inhabited by a strange alien species of rabbits, these rabbits exhibited space-bending properties similar to that of bluespace material but on a much more primitive level. The organization captured and thoroughly studied many rabbit test subjects to explore ways they can harness their abilities for warfare, sure enough scientists were able to craft powerful weapons and gadgets out of their skull and bones. Due to an over abundance of these rabbits, they became the primary meal for the company members on the facility. Ingesting this rabbit however came with severe side effects, many members were driven mad and some even began seeing visions..
</details>

<details>
<summary>Weapons</summary>
<br>

monsterhunter weapons

Some of the prototype weapons crafted with the skull and bones of rabbits, each weapon comes with 2 forms:

Darkmoon Greatsword:
Initially, this weapon is a heavy hilt coated in gold, its very small and compact but can still be used as a weapon
The darkmoon blade can be extended from the hilt, the blade has a ranged attack that charges over 5 seconds and deals alot of burn damage against monsters.

Threaded Cane:
This weapon starts out as a cane but can be extended into a whip. In the cane form, the weapon will do more damage but as a whip it can do damage to monsters over 2 tiles away from you

Hunter's Axe:
Starts out as a smaller single blade axe but can be transformed into a double bladed axe. The single blade form is smaller and does decent damage enough by itself. the double blade form does more damage but needs to be held in both hands.
</details>

<details>
<summary>White Rabbits</summary>
<br>

rabbit lol

There will be several white rabbits scattered throughout the station, however they cannot be seen by regular crewmembers. Only a monster hunter will be able to perceive them. Monster hunters will be given a tool to find them. Finding and Interacting with these rabbits can give the monster hunter various item rewards. They will also give the player hints about his targets; such as what abilities their targets have so that the hunter can better prepare before meeting them.
</details>

<details>
<summary>Upgrading your weapon</summary>
<br>

When you interact with the rabbit, it escapes through a rabbit hole and drops one of its eyes. This eye can be used to upgrade your weapon. For every eye used, your weapon will have an increased 7% damage for both forms  and it can be upgraded up to 3 times.
</details>

<details>
<summary>The Wonderland</summary>
<br>

wonderland

One of the rabbits will drop you an item that'll allow you to drift into the "wonderland". This is essentially a breeding facility created by the monster hunter foundation to house and cultivate rabbit test subjects. In the wonderland, you can upgrade your weapon:

weapon forge

Place your weapon on the table, and then use the rabbit's eye on the table. You will have to be quick about it as you only get a minute to explore the wonderland before you will forcefully be teleported back to the station. You will be able to visit the wonderland once every 10 minutes.
</details>

<details>
<summary>The Rabbit's mask</summary>
<br>
A mask created through grafting several rabbits' faces together. Wearing this mask will allow you to teleport into the wonderland and will also give you a new jaunt ability: the paradox rabbit. This is a jaunt ability BUT with a twist. Upon activating this ability, you will be transformed into a rabbit inside a small pocket dimension, moving this rabbit will also move your human character on the station, allowing you to bypass any walls or objects in between. For example, if you move the rabbit 2 steps to the right, your character will move 2 steps to the right. Here is an example of me using it to escape from prison:

https://user-images.githubusercontent.com/95004236/209896968-a6c52d9f-68a5-4780-8f1f-811c3fa3e32b.mp4

These abilities can only be used only when wearing the mask
</details>

<details>
<summary>Greentexting</summary>
<br>
 So far everything I have described IS side content and not required to greentext. you can choose to ignore the rabbit hunt and KILL monsters the way you prefer. You will be tasked with hunting down 3 different monsters on the station, which can be changelings or vampires. Although not finding the rabbits is required, finding them will reward you with a secret ending.
</details>

<details>
<summary>The Jabberwocky</summary>
<br>
(placeholder sprite)

jabberwocky

Upon killing all your targets, if hunters found all the rabbits on the station their bodies will be transformed into this hideous state. They lose all sanity and are focused on annihilation and destruction of the station and possess several wacky abilities.
Its meant to be a miniboss fight against the crew, similar to the space dragon but not as strong.
</details>

### CODE REVIEW

**Comment by GuillaumePrata** [comment]
wtf, why I love monster hunter now?

File: code/modules/unit_tests/simple_animal_freeze.dm [hunk 1]
```diff
@@ -383,6 +383,8 @@
 		/mob/living/simple_animal/hostile/gorilla/albino,
 		/mob/living/simple_animal/hostile/devil/arch_devil,
 		/mob/living/simple_animal/hostile/devil,
+		/mob/living/simple_animal/hostile/megafauna/red_rabbit,
+		/mob/living/simple_animal/hostile/killer_rabbit
```
Comment (SgtHunk) [inline code/modules/unit_tests/simple_animal_freeze.dm:387]:
```suggestion
		/mob/living/simple_animal/hostile/killer_rabbit,
```

File: fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/hunting_contract.dm [hunk 2]
```diff
@@ -0,0 +1,132 @@
+/obj/item/hunting_contract
+	name = "\improper Hunter's Contract"
+	desc = "Should I have my lawyer read this?"
```
Comment (SgtHunk) [inline fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/hunting_contract.dm:3]: Description should be more objective of the item.

File: fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/hunting_contract.dm [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+/obj/item/hunting_contract/Initialize(mapload, datum/antagonist/monsterhunter/hunter)
+	. = ..()
+	shop = new /datum/hunter_market
+	if(hunter)
+		owner = hunter
+
+/obj/item/hunting_contract/ui_interact(mob/living/user, datum/tgui/ui)
+	if(!IS_MONSTERHUNTER(user))
+		to_chat(usr, span_notice("What the hell do these symbols mean?"))
```
Comment (SgtHunk) [inline fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/hunting_contract.dm:24]: Same as above, be a bit more objective with it. "You are unable to decipher the symbols." as an example

File: fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/hunting_contract.dm [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+		if("select")
+			if(isnull(params["item"]))
+				return
+			var/item = text2path(params["item"])
+			selected_item = item
+			. = TRUE
+			purchase(selected_item, usr)
+		if("claim_reward")
+			if(!is_station_level(usr.loc.z))
+				to_chat(usr,span_warning("The pull of the ice moon isn't strong enough here.."))
```
Comment (SgtHunk) [inline fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/hunting_contract.dm:71]:
```suggestion
				to_chat(usr,span_warning("The pull of the ice moon isn't strong enough here..."))
```

File: fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_abilities.dm [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+/datum/action/cooldown/paradox/New(Target)
+	..()
+	chessmark = GLOB.wonderland_marks["Wonderchess landmark"]
+	landmark =  GLOB.wonderland_marks["Wonderland landmark"]
+
+
+
+/datum/action/cooldown/paradox/Activate()
+	if(!is_station_level(owner.loc.z))
+		to_chat(owner,span_warning("The pull of the ice moon isn't strong enough here.."))
```
Comment (SgtHunk) [inline fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_abilities.dm:24]:
```suggestion
		to_chat(owner,span_warning("The pull of the ice moon isn't strong enough here..."))
```

File: fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_abilities.dm [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+	StartCooldown(360 SECONDS, 360 SECONDS)
+	if(!landmark)
+		return
+	original_loc = get_turf(owner)
+	var/turf/theplace = get_turf(landmark)
+	owner.forceMove(theplace)
+	var/mob/living/sleeper = owner
+	sleeper.Sleeping(2 SECONDS)
+	StartCooldown()
+	//sleep(3 SECONDS)
```
Comment (SgtHunk) [inline fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_abilities.dm:80]: ?

File: fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_abilities.dm [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+	to_chat(owner, span_warning("You wake up in the Wonderland"))
```
Comment (SgtHunk) [inline fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_abilities.dm:81]: Similar to the other text reviews,,, plz give it punctuation and stuff

File: fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_abilities.dm [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+	playsound(owner, 'fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/sounds/wonderlandmusic.ogg',30)
+	addtimer(CALLBACK(src, PROC_REF(return_to_station), owner), 1 MINUTES)
+
+/datum/action/cooldown/wonderland_drop/proc/return_to_station()
+	if(!original_loc)
+		return
+	owner.forceMove(original_loc)
+	to_chat(owner, span_warning("Was it all a dream?"))
```
Comment (SgtHunk) [inline fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_abilities.dm:89]: Same as above, more objectivity in the text

File: fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_abilities.dm [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+	original_loc = null
+
+/datum/action/cooldown/spell/conjure_item/blood_silver
+	name = "Create bloodsilver bullet"
+	desc = "Wield your blood and mold it into a bloodsilver bullet"
```
Comment (SgtHunk) [inline fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_abilities.dm:94]: SAME AS ABOVE

File: fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_datum.dm [hunk 10]
```diff
@@ -161,3 +200,69 @@
...
+	owner.transfer_to(evil_rabbit)
+	var/datum/objective/survive/destruction = new
+	destruction.name = "Wreak Havoc"
+	destruction.explanation_text = "Wreak havoc upon the station"
+	destruction.owner = owner
+	objectives += destruction
+	for(var/obj/machinery/power/apc/apc as anything in GLOB.apcs_list)
+		if(is_station_level(apc.z))
+			apc.overload_lighting()
+	priority_announce("Whh@t the h?!l is going on?! WEeE have detected a massive upspike in %^%*&^%$! c())@ming from your st!*i@n! GeEeEEET out of TH3RE NOW!!","?????????", 'fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/sounds/beastification.ogg')
```
Comment (SgtHunk) [inline fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_datum.dm:259]: Announcement could maybe look better tbh

File: fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/red_rabbit.dm [hunk 11]
```diff
@@ -0,0 +1,159 @@
+/mob/living/simple_animal/hostile/megafauna/red_rabbit
+	name = "Jabberwocky"
```
Comment (SgtHunk) [inline fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/red_rabbit.dm:2]:
```suggestion
	name = "jabberwocky"
```

File: fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/red_rabbit.dm [hunk 12]
```diff
@@ -0,0 +1,159 @@
...
+	for(var/i in 1 to 3 )
+		var/mob/living/simple_animal/hostile/killer_rabbit/rabbit = new /mob/living/simple_animal/hostile/killer_rabbit(owner.loc)
+		rabbit.GiveTarget(target)
+		rabbit.faction = owner.faction.Copy()
+	StartCooldown()
+
+
+
+/mob/living/simple_animal/hostile/killer_rabbit
+	name = "Killer baby rabbit"
```
Comment (SgtHunk) [inline fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/red_rabbit.dm:73]:
```suggestion
	name = "killer baby rabbit"
```

File: fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/red_rabbit.dm [hunk 13]
```diff
@@ -0,0 +1,159 @@
...
+	melee_damage_upper = 5
+
+/mob/living/simple_animal/hostile/killer_rabbit/AttackingTarget()
+	var/mob/living/carbon/human
+	if(iscarbon(target))
+		human = target
+	else
+		return
+	if(human)
+		explosion(src,heavy_impact_range = 1, light_impact_range = 1, flame_range = 2)
```
Comment (SgtHunk) [inline fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/red_rabbit.dm:88]: what the hell this bitch causing hull breaches !!!

File: fulp_modules/features/antagonists/bloodsuckers/code/rulesets/monsterhunter_event.dm [hunk 14]
```diff
@@ -1,50 +1,12 @@
...
-			continue
-		if(all_players.mind.assigned_role.departments_bitflags & (DEPARTMENT_BITFLAG_SECURITY|DEPARTMENT_BITFLAG_COMMAND))
-			continue
-		if(is_special_character(all_players))
-			continue
-		if(!all_players.getorgan(/obj/item/organ/internal/brain))
-			continue
-		all_players.mind.add_antag_datum(/datum/antagonist/monsterhunter)
-		message_admins("MONSTERHUNTER NOTICE: [all_players] has awoken as a Monster Hunter.")
-		break
```
Comment (SgtHunk) [inline fulp_modules/features/antagonists/bloodsuckers/code/rulesets/monsterhunter_event.dm:1]:
```suggestion
```

**Review by SgtHunk — CHANGES_REQUESTED** [review]
cool antag

File: fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/red_rabbit.dm (hunk 13 above)
Comment (SgtHunk) [reply fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/red_rabbit.dm:88]: also look at indents on L85-L86 lol

File: fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/hunting_contract.dm [hunk 15]
```diff
@@ -0,0 +1,132 @@
...
+	desc = "simple axe for hunters that lean towards barbarian tactics, can transform into a double bladed axe."
+	item = /obj/item/melee/trick_weapon/hunter_axe
+
+/datum/hunter_weapons/darkmoon_greatsword
+	name = "Darkmoon greatsword"
+	desc = "a heavy sword hilt that would knock anyone out cold, can transform into the darkmoonlight greatsword. "
+	item = /obj/item/melee/trick_weapon/darkmoon
+
+
+/obj/item/hunting_contract/Destroy()
```
Comment (JohnFulpWillard) [inline fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/hunting_contract.dm:129]: Destroy should be above all the subtypes

File: fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_abilities.dm [hunk 16]
```diff
@@ -0,0 +1,107 @@
...
+	item_type = /obj/item/ammo_casing/silver
+	spell_requirements = NONE
+	delete_old = FALSE
+
+/datum/action/cooldown/spell/blood_silver/conjure_item/cast(mob/living/carbon/cast_on)
+	if(cast_on.blood_volume < BLOOD_VOLUME_NORMAL)
+		to_chat(cast_on, span_warning ("Using this ability would put our health at risk!"))
+		return
+	. = ..()
+	cast_on.blood_volume -= 20
```
Comment (JohnFulpWillard) [inline fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_abilities.dm:107]: Why are we using blood volume for these abilities? Especially as a 2 minute cooldown ability, this doesn't do anything to a monster hunter.

File: fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_datum.dm [hunk 17]
```diff
@@ -15,6 +15,10 @@
 	var/give_objectives = TRUE
 	var/datum/action/bloodsucker/trackvamp = new /datum/action/bloodsucker/trackvamp()
 	var/datum/action/bloodsucker/fortitude = new /datum/action/bloodsucker/fortitude/hunter()
```
Comment (JohnFulpWillard) [inline fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_datum.dm:17]: Should fortitude and the martial art be part of their uplink?

File: fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_datum.dm [hunk 18]
```diff
@@ -37,44 +41,79 @@
...
-		var/datum/objective/bloodsucker/monsterhunter/monsterhunter_objective = new
-		monsterhunter_objective.owner = owner
-		objectives += monsterhunter_objective
-		//Give Theft Objective
-		var/datum/objective/steal/steal_objective = new
-		steal_objective.owner = owner
-		steal_objective.find_target()
-		objectives += steal_objective
+		find_monster_targets()
+	var/datum/map_template/wonderland/wonder = new()
```
Comment (JohnFulpWillard) [inline fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_datum.dm:47]: You should add a check to not load several if there's multiple MHs.

File: fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_datum.dm [hunk 19]
```diff
@@ -15,6 +15,10 @@
 	var/give_objectives = TRUE
 	var/datum/action/bloodsucker/trackvamp = new /datum/action/bloodsucker/trackvamp()
 	var/datum/action/bloodsucker/fortitude = new /datum/action/bloodsucker/fortitude/hunter()
+	///the rabbit illusion trauma related to us
+	var/datum/brain_trauma/special/rabbit_hole/sickness
```
Comment (JohnFulpWillard) [inline fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_datum.dm:19]: This sells a monster hunter out, right?
Is it intentional? Can monster hunters be "cured"?

File: fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_datum.dm [hunk 20]
```diff
@@ -37,44 +41,79 @@
...
+	killer.gain_trauma(disease)
+	sickness = disease
+	var/mob/living/carbon/criminal = owner.current
+	var/obj/item/rabbit_locator/card = new(criminal,src)
+	var/list/slots = list ("backpack" = ITEM_SLOT_BACKPACK)
+	criminal.equip_in_one_of_slots(card, slots)
+	var/obj/item/hunting_contract/contract = new(criminal,src)
+	criminal.equip_in_one_of_slots(contract, slots)
+	RegisterSignal(src, GAIN_INSIGHT, .proc/insight_gained)
+	RegisterSignal(src, BEASTIFY, .proc/turn_beast)
```
Comment (JohnFulpWillard) [inline fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_datum.dm:66]: Signals should at the very least start with COMSIG, you also should generally try to name it after where the signal is. (``BEASTIFY`` doesn't really tell me when this signal is sent)

File: fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_datum.dm [hunk 21]
```diff
@@ -161,3 +200,75 @@
...
+		var/datum/objective/assassinate/kill_monster = new
+		kill_monster.owner = owner
+		var/datum/mind/target = pick(possible_targets)
+		possible_targets -= target
+		kill_monster.target = target
+		kill_monster.update_explanation_text()
+		objectives += kill_monster
+
+
+/datum/antagonist/monsterhunter/proc/turn_beast()
```
Comment (JohnFulpWillard) [inline fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_datum.dm:251]: Is this something you really want to add? Monster Hunters are meant to be a human that gains power by learning FROM monsters but will not perform as efficient as them, turning them into one seems to contradict that.

If you're willing to allow, I have some ideas for alternatives that I can maybe code myself.

File: fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_datum.dm [hunk 22]
```diff
@@ -161,3 +200,75 @@
...
+
+	var/mob/living/simple_animal/hostile/megafauna/red_rabbit/evil_rabbit = new (get_turf(owner.current))
+	owner.current.gib()
+	owner.transfer_to(evil_rabbit)
+	var/datum/objective/survive/destruction = new
+	destruction.name = "Wreak Havoc"
+	destruction.explanation_text = "Wreak havoc upon the station"
+	destruction.owner = owner
+	objectives += destruction
+	for(var/obj/machinery/power/apc/apc as anything in GLOB.apcs_list)
```
Comment (JohnFulpWillard) [inline fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_datum.dm:262]: Do we want EVERY single apc to overload? I don't think this happens at any other time in the game, even in lategame cult nar'sie-level shit. Seems more annoying than anything, and doesn't make much sense either.

File: fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_datum.dm [hunk 23]
```diff
@@ -161,3 +200,75 @@
...
+	SIGNAL_HANDLER
+
+	var/mob/living/simple_animal/hostile/megafauna/red_rabbit/evil_rabbit = new (get_turf(owner.current))
+	owner.current.gib()
+	owner.transfer_to(evil_rabbit)
+	var/datum/objective/survive/destruction = new
+	destruction.name = "Wreak Havoc"
+	destruction.explanation_text = "Wreak havoc upon the station"
+	destruction.owner = owner
+	objectives += destruction
```
Comment (JohnFulpWillard) [inline fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_datum.dm:261]: This should be it's own objective I think

File: fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_weapons.dm [hunk 24]
```diff
@@ -0,0 +1,481 @@
...
+	///is the weapon in its transformed state?
+	var/enabled = FALSE
+
+
+/obj/item/melee/trick_weapon/proc/upgrade_weapon()
+	SIGNAL_HANDLER
+
+	upgrade_level++
+	force = upgraded_val(base_force,upgrade_level)
+	var/datum/component/transforming/transform = src.GetComponent(/datum/component/transforming)
```
Comment (JohnFulpWillard) [inline fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_weapons.dm:26]:
```suggestion
	var/datum/component/transforming/transform = GetComponent(/datum/component/transforming)
```

File: fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_weapons.dm [hunk 25]
```diff
@@ -0,0 +1,481 @@
...
+	attack_verb_simple = list("attack", "slash", "stab", "slice", "tear", "lacerate", "rip", "dice", "cut")
+	///ready to launch a beam attack?
+
+	COOLDOWN_DECLARE(moonbeam_fire)
+
+
+/obj/item/melee/trick_weapon/darkmoon/Initialize(mapload)
+	. = ..()
+	force = base_force
+	AddComponent(/datum/component/transforming, \
```
Comment (JohnFulpWillard) [inline fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_weapons.dm:61]: Why not add a tranforming trick weapon subtype to avoid this copy paste?

File: fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_weapons.dm [hunk 26]
```diff
@@ -0,0 +1,481 @@
...
+	on_force = 20
+	base_force = 17
+	throwforce = 12
+	damtype = BURN
+	hitsound = 'sound/weapons/bladeslice.ogg'
+	attack_verb_continuous = list("attacks", "slashes", "stabs", "slices", "tears", "lacerates", "rips", "dices", "cuts")
+	attack_verb_simple = list("attack", "slash", "stab", "slice", "tear", "lacerate", "rip", "dice", "cut")
+	///ready to launch a beam attack?
+
+	COOLDOWN_DECLARE(moonbeam_fire)
```
Comment (JohnFulpWillard) [inline fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_weapons.dm:55]:
```suggestion

	///ready to launch a beam attack?
	COOLDOWN_DECLARE(moonbeam_fire)
```

File: fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_weapons.dm [hunk 27]
```diff
@@ -0,0 +1,481 @@
...
+		return
+	if(man.has_movespeed_modifier(/datum/movespeed_modifier/silver_bullet))
+		return
+	if(!(IS_BLOODSUCKER(man)) && !(man.mind.has_antag_datum(/datum/antagonist/changeling)))
+		return
+	man.add_movespeed_modifier(/datum/movespeed_modifier/silver_bullet)
+	addtimer(CALLBACK(man, /mob/living/carbon.proc/remove_bloodsilver), 20 SECONDS)
+
+/mob/living/carbon/proc/remove_bloodsilver()
+	if (src.has_movespeed_modifier(/datum/movespeed_modifier/silver_bullet))
```
Comment (JohnFulpWillard) [inline fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_weapons.dm:277]:
```suggestion
	if (has_movespeed_modifier(/datum/movespeed_modifier/silver_bullet))
```
Why is this a carbon proc? why not just call ``remove_movespeed_modifier`` when you need?

File: fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_weapons.dm [hunk 28]
```diff
@@ -0,0 +1,481 @@
...
+	for(var/obj/effect/located as anything in mental.white_rabbits)
+		if(get_dist(user,located) < dist)
+			dist = get_dist(user,located)
+	return dist
+
+/obj/item/rabbit_locator/Destroy()
+	if(mental)
+		mental.locator = null
+		mental = null
+	. = ..()
```
Comment (JohnFulpWillard) [inline fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_weapons.dm:434]:
```suggestion
	return ..()
```

**Review by JohnFulpWillard — CHANGES_REQUESTED** [review]
1. You add way too many newlines between things, 1-2 generally is fine. This isn't big but it bugged me while reading through this.
2. Please split monster hunter code into it's own folder (like infiltrator) rather than being in the Bloodsucker one, Monster Hunters no longer revolve around Bloodsuckers as their primary victims.

File: fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_datum.dm (hunk 18 above)
Comment (SmoSmoSmoSmok) [reply fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_datum.dm:47]: only 1 can spawn

File: fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_weapons.dm (hunk 25 above)
Comment (SmoSmoSmoSmok) [reply fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_weapons.dm:61]: not all the trick weapons use this component (axe)

File: fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_datum.dm (hunk 22 above)
Comment (SmoSmoSmoSmok) [reply fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_datum.dm]: ma
